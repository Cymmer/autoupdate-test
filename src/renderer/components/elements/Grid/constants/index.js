export { default as gridTypes } from './gridTypes';
