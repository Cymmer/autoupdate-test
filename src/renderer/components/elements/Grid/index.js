import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

import gridTypes from './constants/gridTypes';

const Grid = React.forwardRef(
  ({ className, type, rowGap, children, ...rest }, ref) => (
    <div
      ref={ref}
      className={cn(
        className,
        styles[`Grid___${type}`],
        styles[`Grid___rowGap___${rowGap}`]
      )}
      {...rest}
    >
      {children}
    </div>
  )
);

Grid.displayName = 'Grid';

Grid.defaultProps = {
  className: null,
  type: gridTypes.THREE,
  rowGap: 16,
};

Grid.propTypes = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  type: PropTypes.oneOf([
    gridTypes.ONE,
    gridTypes.TWO,
    gridTypes.THREE,
    gridTypes.FOUR,
  ]),

  // the value for the gap between rows
  rowGap: PropTypes.oneOf([8, 16]),

  // these are all the other props passed down from
  // react-beautiful-dnd's Droppable component
  rest: PropTypes.object,
};

export default Grid;
