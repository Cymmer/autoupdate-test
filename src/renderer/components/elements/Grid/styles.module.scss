@use '../../../scss' as *;

.Grid {
  &___one {
    display: grid;
    grid-template-columns: 1fr;
  }

  &___two {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 24px;

    @include breakpoint(sm) {
      grid-template-columns: 1fr;
    }
  }

  &___three {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-column-gap: 24px;
    width: 100%;

    @include breakpoint(md) {
      grid-template-columns: 1fr 1fr;
    }

    @include breakpoint(sm) {
      grid-template-columns: 1fr;
    }
  }

  &___four {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-column-gap: 24px;
    width: 100%;

    @include breakpoint(md) {
      grid-template-columns: 1fr 1fr;
    }

    @include breakpoint(xs) {
      grid-template-columns: 1fr;
    }
  }

  &___rowGap {
    &___8 {
      grid-row-gap: 8px;
    }

    &___16 {
      grid-row-gap: 16px;
    }
  }
}
