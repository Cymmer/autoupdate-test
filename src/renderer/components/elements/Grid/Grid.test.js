import React from 'react';
import { shallow } from 'enzyme';
import Grid from './index';

describe('RENDERING - Grid', () => {
  it('should render with default props', () => {
    const component = shallow(<Grid>Test</Grid>);
    expect(component).toMatchSnapshot();
  });
});
