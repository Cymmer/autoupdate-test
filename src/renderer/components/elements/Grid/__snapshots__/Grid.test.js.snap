// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Grid should render with default props 1`] = `
<div
  className="Grid___three Grid___rowGap___16"
>
  Test
</div>
`;
